webapp/app/context-api/usersContext.js

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+export const users = {
+  data: [],
+  selectedUser: {},
+  fetch: {
+    loading: false,
+    error: null
+  },
+  getUsers: () => {},
+  createUser: () => {},
+  updateUser: () => {},
+  deleteUser: () => {},
+  selectUser: () => {}
+};
+
+export const UsersContext = React.createContext({
+  ...users
```

Comment:
Spread operator is not cloning the `fetch `object, it's values are passed by reference, the spread is one level deep cloning, so you are unexpectedly mutating the `users` const whem the fetch fields are updated, I would better use lodash cloneDeep.